pom.xml

```diff
@@ -558,6 +563,11 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
+            <version>4.1</version>
+        </dependency>
```

Comment:
Adding a different dependency to deal with the guava dependency.  Must we?  

Please at least put PR comments pointing to why each of these are added.

Reply:
well I can get rid of this

Reply:
Unless there's a big advantage, I'd rather we get rid of it.  If we keep it we should use the most recent version (v4.4), right?

Reply:
there are some convenient helpers for Iterable (I just don't want to reinvent the wheel..).
BTW not used in core so we avoid conflicts. Yes last version would be better. And I trust ASF commons folks to keep backward compat..